### PR TITLE
Update timelinejs auto-update config & repository url

### LINKED ISSUE #10540: [Request] Add TimelineJS3
**Library name:** TimelineJS3
**Git repository url:** https://github.com/NUKnightLab/TimelineJS3
**License(s):** MPL-2.0
**Official homepage:** http://timeline.knightlab.com/

### PR DESCRIPTION
**Pull request by clairetsai818**
close #10540. The old GitHub repo is currently out of maintainence and moved to a new repo.
@cdnjs/intern3 Please help me review this PR, thanks!

Pull request for issue: #10540 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):https://github.com/NUKnightLab/TimelineJS3
 * Official website (optional, not the repository): http://timeline.knightlab.com/
 * License and its reference: [MPL-2.0](https://github.com/NUKnightLab/TimelineJS3/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 850
   - Count of stars: 950
   - Count of forks: 253

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/timelinejs/package.json [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
     "timeline"
   ],
+  "license": "MPL-2.0",
   "autoupdate": {
     "source": "git",
-    "target": "git://github.com/NUKnightLab/TimelineJS.git",
+    "target": "git://github.com/NUKnightLab/TimelineJS3.git",
     "fileMap": [
       {
         "basePath": "build",
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Wrong basePath

File: ajax/libs/timelinejs/package.json [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
         "files": [
-          "css/**/*",
-          "js/**/*"
+          "compiled/css/timeline*",
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Should include all file under `css/` and `js/` cause files are used in `.css` and `.js` files

File: ajax/libs/timelinejs/package.json [hunk 3]
```diff
@@ -1,30 +1,30 @@
 {
   "name": "timelinejs",
   "filename": "js/timeline-min.js",
-  "version": "2.36.0",
+  "version": "3.4.7",
   "description": "TimelineJS is an open-source tool that enables you to build visually-rich interactive timelines and is available in 40 languages.",
-  "homepage": "http://timeline.knightlab.com/",
+  "homepage": "https://timeline.knightlab.com/",
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Separate the commit message with updating auto-update config.

**Comment by extend1994** [comment]
Besides, the files like `timeline-embed.js` should be minified and maybe we should discuss with the mentor that if "new version" of timelinejs could be conbined with the existed library on the cdnjs cause it's a rewrite veriosn ([ref](https://github.com/NUKnightLab/TimelineJS3#overview))

**Comment by clairetsai818** [comment]
@extend1994 I've ask our mentor that it's okay to combined this library with the existing library in this case, and I separate the commit, please help me review this again, thanks!

File: ajax/libs/timelinejs/package.json [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
     "timeline"
   ],
   "autoupdate": {
     "source": "git",
-    "target": "git://github.com/NUKnightLab/TimelineJS.git",
+    "target": "git://github.com/NUKnightLab/TimelineJS3.git",
     "fileMap": [
       {
-        "basePath": "build",
+        "basePath": "",
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Use `compiled` as `basePath`

File: ajax/libs/timelinejs/package.json [hunk 5]
```diff
@@ -1,28 +1,30 @@
...
         "files": [
-          "css/**/*",
-          "js/**/*"
+          "compiled/css/**/*",
+          "compiled/css/*",
+          "compiled/js/**/*",
+          "compiled/js/*"
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: We only need `css/**/*` and `js/**/*`

File: ajax/libs/timelinejs/package.json [hunk 6]
```diff
@@ -1,28 +1,30 @@
...
         ]
       }
     ]
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/NUKnightLab/TimelineJS.git"
+    "url": "https://github.com/NUKnightLab/TimelineJS3.git"
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Maybe this one should also separate to another commit?

File: ajax/libs/timelinejs/package.json [hunk 7]
```diff
@@ -1,28 +1,30 @@
 {
   "name": "timelinejs",
-  "filename": "js/timeline-min.js",
-  "version": "2.36.0",
+  "filename": "js/timeline.min.js",
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: the file `timeline-min.js` is what we need and the minified file of `timeline.js`.

File: ajax/libs/timelinejs/package.json (hunk 6 above)
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: This one can be combined with the auto-update commit cause it's a part of the version update. Ref: https://github.com/cdnjs/cdnjs/pull/10416#discussion_r102888929

**Comment by clairetsai818** [comment]
@PeterDaveHello I deleted them accidentally, and I got those files back and revised the first commit message

**Comment by PeterDaveHello** [comment]
@clairetsai818 you should mention the library name you are working on in the first commit message.

**Comment by clairetsai818** [comment]
@PeterDaveHello I revised the commit message, please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
hmmm where is @sufuf3 or @hare1039 ?

**Comment by PeterDaveHello** [comment]
1. Wrap your commit message, again.
2. Try to use single line commit message to make the things to be clear enough like:
`Update timelinejs homepage url to use https`

**Comment by PeterDaveHello** [comment]
@cdnjs/intern3 don't forget to check about if the commit message needs to be wrapped or not.

**Comment by clairetsai818** [comment]
@PeterDaveHello I revised the commit message, please help me review this PR again. Thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 please separate the update of auto-update config into another commit, thanks.

**Comment by maruilian11** [comment]
Should delete the auto-update in the second commit.

**Comment by PeterDaveHello** [comment]
@maruilian11 can you please explain in detail?

**Comment by maruilian11** [comment]
in `Add git auto-update config to timelinejs`

**Comment by PeterDaveHello** [comment]
@maruilian11 so what's the reason to delete it?

**Comment by maruilian11** [comment]
sorry, I didn't express it well. I mean update the auto-update in `Add git auto-update config to timelinejs`.

**Comment by clairetsai818** [comment]
@PeterDaveHello please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
Please also add `cc #10540` in the third commit message.

**Comment by PeterDaveHello** [comment]
Where is your `close #10540` commit message?

File: ajax/libs/timelinejs/package.json [hunk 8]
```diff
@@ -1,29 +1,29 @@
...
           "css/**/*",
           "js/**/*"
         ]
       }
     ]
   },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/NUKnightLab/TimelineJS.git"
-  },
```
Comment (PeterDaveHello) [inline ajax/libs/timelinejs/package.json]: What's the reason why to move repository upper?

Comment (clairetsai818) [inline ajax/libs/timelinejs/package.json]: Should I move `repository` down? It's my mistake, there is no reason to move them up.

Comment (PeterDaveHello) [inline ajax/libs/timelinejs/package.json]: Yes let's move it back, we should not touch unrelated things, the less diff the better!

**Comment by clairetsai818** [comment]
@PeterDaveHello I've moved the `repository` field back, please help me review this, thanks!

**Comment by maruilian11** [comment]
why not combine these two commits?

File: ajax/libs/timelinejs/package.json [hunk 9]
```diff
@@ -1,28 +1,15 @@
 {
   "name": "timelinejs",
   "filename": "js/timeline-min.js",
-  "version": "2.36.0",
+  "version": "3.4.7",
   "description": "TimelineJS is an open-source tool that enables you to build visually-rich interactive timelines and is available in 40 languages.",
   "homepage": "https://timeline.knightlab.com/",
   "keywords": [
     "timeline"
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Add some keywords to it. Ref: https://github.com/NUKnightLab/TimelineJS/blob/master/bower.json

File: ajax/libs/timelinejs/package.json [hunk 10]
```diff
@@ -1,28 +1,15 @@
...
   ],
-  "autoupdate": {
```
Comment (extend1994) [inline ajax/libs/timelinejs/package.json]: Here, I think you should not delete or change the original auto-update config until you want to update it.

**Comment by clairetsai818** [comment]
@maruilian11 Could you please tell me which two commits? Thanks!

**Comment by PeterDaveHello** [comment]
Could also add them first.

**Comment by clairetsai818** [comment]
I rebased this branch, @PeterDaveHello please help me review this, thanks!

**Comment by clairetsai818** [comment]
@PeterDaveHello I revised the commit msg, please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 ask review from the others first.

File: ajax/libs/timelinejs/package.json (hunk 9 above)
Comment (maruilian11) [inline ajax/libs/timelinejs/package.json]: @clairetsai818 why not enrich the keywords?

**Comment by clairetsai818** [comment]
@maruilian11 Do you mean I should let the robot to add v3.4.7?

**Comment by maruilian11** [comment]
Yes, we only add the latest version for new lib, except lib with filename modifying.

**Comment by clairetsai818** [comment]
@maruilian11 I removed v3.4.7, please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
yeah I can take care of the old versions here.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by clairetsai818** [comment]
@PeterDaveHello this Pr is waiting for the author's response

**Comment by clairetsai818** [comment]
@maruilian11 About the commented out files:https://github.com/NUKnightLab/TimelineJS3/issues/490
It's actually not all comments if you see its raw files.

**Comment by PeterDaveHello** [comment]
Author's review?

**Comment by clairetsai818** [comment]
@PeterDaveHello I edited the comment, please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
The second commit message should be improved.

**Comment by PeterDaveHello** [comment]
The second commit message should be improved.

**Comment by PeterDaveHello** [comment]
btw IMO it's fine to add the latest version if there are not too many files.
